common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java

```diff
@@ -242,7 +240,6 @@ public ShuffleMetrics() {
       allMetrics.put("registeredExecutorsSize",
                      (Gauge<Integer>) () -> blockManager.getRegisteredExecutorsSize());
       allMetrics.put("numActiveConnections", activeConnections);
-      allMetrics.put("numRegisteredConnections", registeredConnections);
```

Comment:
Ur, this might remove this metric from the other resource managers like `Mesos`. Are you sure about the side-effect?

Reply:
If you need to fix some YARN issue, I'd like to recommend to remove from YARN side and leave this shared class AS-IS.

Reply:
This line looks suspicious and confusing since `numRegisteredConnections` is registered twice for `YarnShuffleService`. After digging into it, I find it's [the Counter from TransportContext](https://github.com/apache/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/TransportContext.java#L71) that's really [counting the numbers](https://github.com/apache/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/server/TransportChannelHandler.java#L189). That created here in `ExternalBlockHandler#ShuffleMetrics` is never used anywhere.

I think it's misleading for people who want to add new metrics. That said, I'm ok to leave it as is if you still suggest so.

Reply:
If you are worried about the side-effect, I can add UTs for all resource managers to ensure all expected metrics are registered. Then maybe it's worth a separate PR. What do you think ?

Reply:
Please create a removal PR first for the following. That will make the review easier~
> After digging into it, I find it's the Counter from TransportContext that's really counting the numbers. That created here in ExternalBlockHandler#ShuffleMetrics is never used anywhere.

Reply:
We will review there if that is really never exposed to any one.

Reply:
@dongjoon-hyun  https://github.com/apache/spark/pull/28457 has been created. Please help review, thanks.